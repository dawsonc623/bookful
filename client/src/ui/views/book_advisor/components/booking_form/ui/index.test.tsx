import React from "react";

import {
  render
} from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import BookingForm  from ".";

import advisorAvailabilityCollectionFactory from "../../../../../../app/advisor_availability_collection/factory";

const noop  = () => {/* No-op*/};

describe("BookingForm", () => {
  test("Renders without error", () => {
    const advisorAvailabilityCollection = advisorAvailabilityCollectionFactory.construct([]);

    const {
      asFragment
    } = render(
      <BookingForm
        advisorAvailability = {advisorAvailabilityCollection}
        bookAdvisor         = {noop}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});