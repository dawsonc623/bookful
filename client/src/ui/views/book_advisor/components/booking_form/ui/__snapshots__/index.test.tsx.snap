// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingForm Renders without error 1`] = `
<DocumentFragment>
  <div
    class="bookingForm"
  >
    <div>
      <hr />
    </div>
    <div
      class="availabilityTable mdc-data-table rmwc-data-table--sticky-rows rmwc-data-table--sticky-rows-1"
    >
      <table
        class="mdc-data-table__table"
      >
        <thead
          class="rmwc-data-table__head"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__header-row"
          >
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Advisor ID
            </th>
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Availabilities
            </th>
            <th
              class="rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="mdc-data-table__content"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__row"
          >
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            />
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              No Availability
            </td>
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
